libraries/azure-app-configuration-importer/src/internal/parsers/defaultConfigurationSettingsConverter.ts

```diff
@@ -262,14 +277,224 @@ class FeatureFlagConfigurationSettingsConverter implements ConfigurationSettings
+          configuration_value: configuration_value
+        });
+      }
+    }
+
+    return featureFlagValue;
+  }
+
+  private isValidUserAllocation(user: object): user is User {
+    const lowerCaseUser = this.getLowerCaseProperties(user);
+    return (
+      typeof lowerCaseUser === "object" &&
+      lowerCaseUser !== null &&
+      typeof lowerCaseUser.variant === "string" &&
+      Array.isArray(lowerCaseUser.users)
+    );
+  }
+
+  private isValidGroupAllocation(group: object): group is Group {
+    const lowerCaseGroup = this.getLowerCaseProperties(group);
+    return (
+      typeof lowerCaseGroup === "object" &&
+      lowerCaseGroup !== null &&
+      typeof lowerCaseGroup.variant === "string" &&
+      Array.isArray(lowerCaseGroup.groups)
+    );
+  }
+
+  private isValidPercentileAllocation(percentile: object): percentile is Percentile {
+    const lowerCasePercentile = this.getLowerCaseProperties(percentile);
+    return (
+      typeof lowerCasePercentile === "object" &&
+      lowerCasePercentile !== null &&
+      typeof lowerCasePercentile.variant === "string" && 
+      lowerCasePercentile.to === "string" && 
+      lowerCasePercentile.from === "string"
+    );
+  }
+
+  private isValidVariant(variant: any): variant is Variant {
+    const status_override = variant[this.featureFlagsKeyWordsCase.status_override];
+    return (
+      typeof variant === "object" &&
+      variant !== null &&
+      typeof variant.name === "string" &&
+      (status_override === undefined ||
+        Object.values(StatusOverride).includes(status_override))
+    );
+  }
+
+  private validateClientFilters(featureFlagName: string, schemaVersion: string, clientFilters: any[]): ClientFilter[] {
+    let validatedClientFilters: ClientFilter[] = [];
+
+    for (let i = 0; i < clientFilters.length; i++) {
+      const parameters: Partial<ClientFilter> = {};
+      //
+      // Converting client_filter keys to lower case
+      const lowerCaseFilters = this.getLowerCaseProperties(clientFilters[i]);
+
+      const filtersName = lowerCaseFilters["name"];
+      const filtersParameters = lowerCaseFilters["parameters"];
+
+      if (filtersName) {
+        if (filtersName.toLowerCase() == "alwayson" && schemaVersion == "ffv1") {
+          validatedClientFilters = [];
+          break;
+        }
+        parameters.name = filtersName;
+        if (filtersParameters) {
+          parameters.parameters = filtersParameters;
+        }
+        validatedClientFilters[i] = parameters as ClientFilter;
+      }
+      else {
+        throw new ArgumentError(
+          `This feature flag '${featureFlagName}' has a filter without the required 'name' property.`
+        );
+      }
+    }
+
+    return validatedClientFilters;
+  }
+
+  private checkIfUsingFFV2Schema(config: object): boolean {
+    let foundFeatureFlagsKey = false;
+
+    for (const key in config) {
+      if (key == this.featureFlagsKeyWordsCase.feature_flags) {
```

Comment:
The new MS FM schema only supports this naming case

```
"feature_managemetn": {
     "feature_flags": [
      ]
}
```

It does not support casing like:
```
"FeatureManagement": {
     "FeatureFlags": [
      ]
}

```

Reply:
So we can't pass casing as parameter here when checking if it's using FF v2 schema or not.

Reply:
I missed this, yes schema v2 only allows this case "feature_flags". This has been updated